specfile/sourcelist.py

```diff
@@ -2,12 +2,15 @@
 # SPDX-License-Identifier: MIT
 
 import collections
-from typing import List, Optional, SupportsIndex, overload
+from typing import TYPE_CHECKING, List, Optional, SupportsIndex, overload
 
 from specfile.rpm import Macros
 from specfile.sections import Section
 from specfile.tags import Comments
 
+if TYPE_CHECKING:
+    from specfile.specfile import Specfile
+
```

Comment:
This is to avoid circular import, `Specfile` is only imported when type checking with mypy, never in runtime. I don't think we should use this anywhere else.